apiv2/db_import/importer.py

```diff
@@ -0,0 +1,246 @@
+            funding.import_to_db()
+
+        if not import_runs:
+            continue
+
+        run_cleaner = StaleRunDeletionDBImporter(dataset_id, config)
+        for run in RunDBImporter.get_item(dataset_id, dataset, config):
+            logger.info("Processing Run with prefix %s", run.dir_prefix)
+            run_obj = run.import_to_db()
+            run_id = run_obj.id
+            run_cleaner.mark_as_active(run_obj)
+
+            if import_tiltseries:
+                tiltseries_cleaner = StaleTiltSeriesDeletionDBImporter(run_id, config)
+                tiltseries = TiltSeriesDBImporter.get_item(run_id, run, config)
+                if tiltseries:
+                    tiltseries_obj = tiltseries.import_to_db()
+                    tiltseries_cleaner.mark_as_active(tiltseries_obj)
+                tiltseries_cleaner.remove_stale_objects()
+
+            if not import_tomogram_voxel_spacing:
+                continue
+
+            voxel_spacing_cleaner = StaleVoxelSpacingDeletionDBImporter(run_id, config)
+            for voxel_spacing in TomogramVoxelSpacingDBImporter.get_items(run_id, run, config):
+                voxel_spacing_obj = voxel_spacing.import_to_db()
+
+                if import_tomograms:
+                    tomogram_cleaner = StaleTomogramDeletionDBImporter(voxel_spacing_obj.id, config)
+                    for tomogram in TomogramDBImporter.get_item(voxel_spacing_obj.id, run_id, voxel_spacing, config):
+                        tomogram_obj = tomogram.import_to_db()
+                        tomogram_cleaner.mark_as_active(tomogram_obj)
+
+                        if import_tomogram_authors:
+                            tomogram_authors = TomogramAuthorDBImporter.get_item(tomogram_obj.id, tomogram, config)
+                            tomogram_authors.import_to_db()
+                    tomogram_cleaner.remove_stale_objects()
+
+                if import_annotations:
+                    annotation_cleaner = StaleAnnotationDeletionDBImporter(run_id, config)
+                    for annotation in AnnotationDBImporter.get_item(voxel_spacing_obj.id, voxel_spacing, config):
+                        annotation_obj = annotation.import_to_db()
+                        annotation_cleaner.mark_as_active(annotation_obj)
+
+                        if import_annotation_authors:
+                            annotation_authors = AnnotationAuthorDBImporter.get_item(
+                                annotation_obj.id,
+                                annotation,
+                                config,
+                            )
+                            annotation_authors.import_to_db()
+                    annotation_cleaner.remove_stale_objects()
+
+                voxel_spacing_cleaner.mark_as_active(voxel_spacing_obj)
+
+            voxel_spacing_cleaner.remove_stale_objects()
+
+        run_cleaner.remove_stale_objects()
+        session.commit()
+    session.commit()
+
+
+if __name__ == "__main__":
```

Comment:
nit: is this needed?

Reply:
Yep, it allows us to run the script on the command line or also import its methods into another module